### DISCUSSION TITLE
How do you add automatic releases to wingetbot

### DISCUSSION THREAD

**Discussion by upintheairsheep** [microsoft/winget-pkgs#106262] [Q&A]
WingetBot is an absolute king when it comes to updating packages, but how do you make the way to make the bot check the versions of the programs and update them?

**Comment by Trenly** [comment]
> WingetBot is an absolute king when it comes to updating packages, but how do you make the way to make the bot check the versions of the programs and update them?

wingetbot only scans existing URLs in the repository to see if the packages at them have changed SHA256 hash. If the hash has changed, it tries to update it. There isn't a way to "add" automatic releases to wingetbot, it just happens whenever a vanity URL is used.

A "vanity" URL is a URL that always stays the same, and typically references the latest version. It's not good practice to use a vanity URL for a package unless necessary. When possible, the direct link to a version should be used and versions should be retained as long as the manifest is working, unless there are extenuating circumstances. For example - if a package is hosted at GitHub, it would not be appropriate to use the `/releases/latest` download URL, since the versioned URL is available; but, for something like Facebook Messenger, only a "latest" URL is available, so that is what had to be used - causing wingetbot to (attempt) an update every time a new version of Facebook Messenger is available.

The best way for automatic releases is for publishers to integrate the wingetcreate.exe directly into their CI/CD process rather than relying on existing automation which may not always work and is prone to errors.
